Wrong inference of annotation by boolean comparison

Consider the code below:
```
class errmsg {
  static void str(String argStr) {
    if (argStr == "abc") {
      return;
    }
    char c = argStr.charAt(argStr.length() + 1);
  }
}
```

Run it by typing in the terminal:
javac -processor org.checkerframework.checker.index.IndexChecker errmsg.java

Output:
```
errmsg.java:8: error: [argument.type.incompatible] incompatible types in argument.
char c = argStr.charAt(argStr.length() + 1);
                                       ^
  found   : @LTLengthOf(value={""abc"", "argStr"}, offset={"-2", "-2"}) int
  required: @LTLengthOf(value="argStr", offset={}) int
1 error
```

The error message suggests that argument of the charAt() function is less than BOTH ("abc".length() + 2) and (argStr.length() + 2), but clearly, it is not necessary.
`argStr == "abc"` is a boolean comparison and not assignment, so the checker should not infer argstr as `@LTEqLengthOf("abc")`

--- a/checker/src/main/java/org/checkerframework/checker/index/samelen/SameLenTransfer.java
+++ b/checker/src/main/java/org/checkerframework/checker/index/samelen/SameLenTransfer.java
@@ -226,6 +226,8 @@ public class SameLenTransfer extends CFTransfer {
             CFValue firstValue,
             CFValue secondValue,
             boolean notEqualTo) {
+        // If result is a Regular transfer, then the elseStore is a copy of the then store, that is
+        // created when getElseStore is called.  So do that before refining any values.
         CFStore elseStore = result.getElseStore();
         CFStore thenStore = result.getThenStore();
         CFStore equalStore = notEqualTo ? elseStore : thenStore;

--- a/checker/src/main/java/org/checkerframework/checker/index/samelen/SameLenTransfer.java
+++ b/checker/src/main/java/org/checkerframework/checker/index/samelen/SameLenTransfer.java
@@ -226,7 +226,9 @@ public class SameLenTransfer extends CFTransfer {
             CFValue firstValue,
             CFValue secondValue,
             boolean notEqualTo) {
-        CFStore equalStore = notEqualTo ? result.getElseStore() : result.getThenStore();
+        CFStore elseStore = result.getElseStore();
+        CFStore thenStore = result.getThenStore();
+        CFStore equalStore = notEqualTo ? elseStore : thenStore;
 
         Node firstLengthReceiver = getLengthNodeReceiver(firstNode);
         Node secondLengthReceiver = getLengthNodeReceiver(secondNode);
@@ -240,7 +242,6 @@ public class SameLenTransfer extends CFTransfer {
             refineEq(firstNode, secondNode, equalStore);
         }
 
-        return new ConditionalTransferResult<>(
-                result.getResultValue(), result.getThenStore(), result.getElseStore());
+        return new ConditionalTransferResult<>(result.getResultValue(), thenStore, elseStore);
     }
 }

--- a/checker/tests/index/Issue2420.java
+++ b/checker/tests/index/Issue2420.java
@@ -1,0 +1,17 @@
+import org.checkerframework.checker.index.qual.*;
+import org.checkerframework.common.value.qual.*;
+
+class Issue2420 {
+    static void str(String argStr) {
+        if (argStr.isEmpty()) {
+            return;
+        }
+        if (argStr == "abc") {
+            return;
+        }
+        // :: error: (argument.type.incompatible)
+        char c = "abc".charAt(argStr.length() - 1);
+        // :: error: (argument.type.incompatible)
+        char c2 = "abc".charAt(argStr.length());
+    }
+}

